lib/datastore/query.js

```diff
@@ -75,7 +75,16 @@ function Query(namespace, kinds) {
 }
 
 /**
+ * @param {boolean=} autoPaginateVal - Have pagination handled automatically.
+ *     Default: true.
  * @return {module:datastore/query}
+ *
+ * @example
+ * // Have pagination handled automatically
+ * var autoPaginated = query.autoPaginate();
```

Comment:
I think a richer example is good, so long as it's representative. That is, we don't want to be building mini apps inside the example snippet, but for example:

```javascript
var ancestoryQuery = query.hasAncestor(dataset.key(['Parent', 123]));
```

Might be better as

```javascript
// Create a couple keys ['Person', 1234] and ['Person', 4567] and ['Person', 1234, 'Person', 'child']
// Create a query for kind=Person
// Set the ancestor to be Person 1234
var ancestoryQuery = query.hasAncestor(dataset.key(['Person', 1234]));
// Mention that the result of the query should be ['Person', 1234, 'Person', 'child']
```